### PR TITLE
Attempt to fix "Text file busy" errors (testing and production)

### PR DESCRIPTION
**Pull request by jmchilton**
@martenson Want to update the API tests to target this branch and add a `--external_tmp` flag after the `--db postgres` flag in `run_tests.sh` call.

### CODE REVIEW

**Comment by martenson** [comment]
@jmchilton I am running them manually as jenkins now.

**Comment by jmchilton** [comment]
Doh - just ran only the workflow tests locally with these options and it is the first time I had seen these issues locally :(.

**Comment by martenson** [comment]
@jmchilton you ran it with this PR?

**Comment by martenson** [comment]
@jmchilton that might mean we are in the correct neighborhood

**Comment by mvdbeek** [comment]
Would be awesome if you can fix it, I ran into this when running workflows that produce a large amount of jobs. It seemed to be more frequent under high server load.

**Comment by jmchilton** [comment]
@mvdbeek I assume you are just running regular jobs through a DRM (e.g. SLURM?) in that case and not using Docker at all?

**Comment by jmchilton** [comment]
Ping @natefoo - okay nothing we discussed worked. Any fresh ideas on this problem?

**Comment by mvdbeek** [comment]
@jmchilton indeed, it's slurm in our case. Importantly, slurm jobs also run on the galaxy node.

**Comment by jmchilton** [comment]
Okay I removed the WIP, I think this sync is ugly but should fix everything not just the local tests with Docker. This PR no longer has the `/tmp` hack that @martenson was kind enough to test repeatedly.

**Comment by martenson** [comment]
ping @natefoo

**Comment by martenson** [comment]
@natefoo said: `"sync is fine"`

**Comment by martenson** [comment]
@jmchilton :( https://jenkins.galaxyproject.org/job/docker-api/185/

File: lib/galaxy/jobs/runners/__init__.py [hunk 1]
```diff
@@ -311,6 +311,18 @@ def get_job_file(self, job_wrapper, **kwds):
...
+        with open( path, 'w' ) as f:
+            f.write( contents )
+        os.chmod( path, mode )
+        try:
+            # sync file system to avoid "Text file busy" problems.
+            # These have occurred both in Docker containers and on EC2 clusters
+            # under high load.
+            subprocess.check_call(["sync"])
+        except Exception:
+            pass
```
Comment (nsoranzo) [inline lib/galaxy/jobs/runners/__init__.py:324]: Would it make sense to `log.debug()` the exception before `pass`?

**Pull request merged**